src/tree/updater_fast_hist.cc

```diff
@@ -552,8 +552,8 @@ class FastHistMaker: public TreeUpdater {
       bst_int split_cond = -1;
       // convert floating-point split_pt into corresponding bin_id
       // split_cond = -1 indicates that split_pt is less than all known cut points
-      for (unsigned i = gmat.cut->row_ptr[fid]; i < gmat.cut->row_ptr[fid + 1]; ++i) {
-        if (split_pt == gmat.cut->cut[i]) split_cond = static_cast<bst_int>(i);
+      for (size_t i = gmat.cut->row_ptr[fid]; i < gmat.cut->row_ptr[fid + 1]; ++i) {
```

Comment:
This line and the next operates on quantiles, so let's keep the type as `unsigned`